### PR TITLE
Switch to gdrv

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Merges the gendrv branch to master in this git. There's some small fixes for conflicts in this merge.

Depends on 
https://github.com/OP-TEE/optee_os/pull/702
https://github.com/OP-TEE/optee_client/pull/47
https://github.com/OP-TEE/optee_test/pull/75

### CODE REVIEW

**Comment by jbech-linaro** [comment]
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

File: qemu.mk [hunk 1]
```diff
@@ -146,8 +139,7 @@ filelist-tee: xtest
 	@echo "# OP-TEE device" >> $(GEN_ROOTFS_FILELIST)
 	@echo "dir /lib/modules 755 0 0" >> $(GEN_ROOTFS_FILELIST)
 	@echo "dir /lib/modules/$(call KERNEL_VERSION) 755 0 0" >> $(GEN_ROOTFS_FILELIST)
-	@echo "file /lib/modules/$(call KERNEL_VERSION)/optee.ko $(OPTEE_LINUXDRIVER_PATH)/core/optee.ko 755 0 0" >> $(GEN_ROOTFS_FILELIST)
-	@echo "file /lib/modules/$(call KERNEL_VERSION)/optee_armtz.ko $(OPTEE_LINUXDRIVER_PATH)/armtz/optee_armtz.ko 755 0 0" >> $(GEN_ROOTFS_FILELIST)
+	@echo "file /lib/modules/$(call KERNEL_VERSION)/optee.ko $(LINUX_PATH)/drivers/tee/optee.ko 755 0 0" >> $(GEN_ROOTFS_FILELIST)
```
Comment (pascal-brand-st-dev) [inline qemu.mk]: correct?

File: fvp.mk [hunk 2]
```diff
@@ -166,8 +153,7 @@ filelist-tee:
 	@echo "# OP-TEE device" >> $(GEN_ROOTFS_FILELIST)
 	@echo "dir /lib/modules 755 0 0" >> $(GEN_ROOTFS_FILELIST)
 	@echo "dir /lib/modules/$(call KERNEL_VERSION) 755 0 0" >> $(GEN_ROOTFS_FILELIST)
-	@echo "file /lib/modules/$(call KERNEL_VERSION)/optee.ko $(OPTEE_LINUXDRIVER_PATH)/core/optee.ko 755 0 0" >> $(GEN_ROOTFS_FILELIST)
-	@echo "file /lib/modules/$(call KERNEL_VERSION)/optee_armtz.ko $(OPTEE_LINUXDRIVER_PATH)/armtz/optee_armtz.ko 755 0 0" >> $(GEN_ROOTFS_FILELIST)
+	@echo "file /lib/modules/$(call KERNEL_VERSION)/optee.ko $(LINUX_PATH)/drivers/tee/optee.ko 755 0 0" >> $(GEN_ROOTFS_FILELIST)
```
Comment (pascal-brand-st-dev) [inline fvp.mk]: required?

File: fvp.mk [hunk 3]
```diff
@@ -44,7 +44,7 @@ ARM_TF_EXPORTS ?= \
 ARM_TF_FLAGS ?= \
 	BL32=$(OPTEE_OS_BIN) \
 	BL33=$(EDK2_BIN) \
-	DEBUG=0 \
+	DEBUG=1 \
```
Comment (pascal-brand-st-dev) [inline fvp.mk]: why this change?

**Comment by pascal-brand-st-dev** [comment]
The last commit is a merge. I would prefer a rebase.

File: qemu.mk (hunk 1 above)
Comment (jenswi-linaro) [inline qemu.mk]: I'll update with correct path

File: fvp.mk (hunk 2 above)
Comment (jenswi-linaro) [inline fvp.mk]: Not really, but it's convenient if compiled as a load module instead.

File: fvp.mk (hunk 3 above)
Comment (jenswi-linaro) [inline fvp.mk]: I'll remove this.

**Comment by jenswi-linaro** [comment]
Rebased

**Comment by pascal-brand-st-dev** [comment]
Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

**Comment by pascal-brand-st-dev** [comment]
@jenswi-linaro The R-Bs are missing in the commit message

**Comment by jenswi-linaro** [comment]
Tag applied
